server/test/test.js

```diff
@@ -0,0 +1,665 @@
+        isAdmin: false,
+        password: '@testPassword1',
+        reEnterPassword: '@testPassword1'
+      };
+      chai.request(server)
+        .post('/api/v1/users/')
+        .set({ token: process.env.TEST_TOKEN })
+        .send(user)
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.be.a('object');
+          res.body.should.have.property('errors');
+          res.body.errors.should.be.a('array');
+          res.body.errors[0].should.eql('Username must be between 4-15 characters \n' +
+          'long, it must include at least one lowercase alphabet, and it must consist of \n' +
+          'only underscores,lowercase alphabets, numbers,\n' +
+          'and no spaces in between characters.');
+          done();
+        });
+    });
+
+    it('it should not create a user if the username is greater than 15 characters', (done) => {
+      const user = {
+        username: 'thisisalongusername',
+        email: 'testemail@test.com',
+        isAdmin: false,
+        password: '@testPassword1',
+        reEnterPassword: '@testPassword1'
+      };
+      chai.request(server)
+        .post('/api/v1/users/')
+        .set({ token: process.env.TEST_TOKEN })
+        .send(user)
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.be.a('object');
+          res.body.should.have.property('errors');
+          res.body.errors.should.be.a('array');
+          res.body.errors[0].should.eql('Username must be between 4-15 characters \n' +
+          'long, it must include at least one lowercase alphabet, and it must consist of \n' +
+          'only underscores,lowercase alphabets, numbers,\n' +
+          'and no spaces in between characters.');
+          done();
+        });
+    });
+
+    it('it should not create a user if the username consists of special characters \n' +
+    'other than an underscore', (done) => {
+      const user = {
+        username: '@test',
+        email: 'testemail@test.com',
+        isAdmin: false,
+        password: '@testPassword1',
+        reEnterPassword: '@testPassword1'
+      };
+      chai.request(server)
+        .post('/api/v1/users/')
+        .set({ token: process.env.TEST_TOKEN })
+        .send(user)
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.be.a('object');
+          res.body.should.have.property('errors');
+          res.body.errors.should.be.a('array');
+          res.body.errors[0].should.eql('Username must be between 4-15 characters \n' +
+          'long, it must include at least one lowercase alphabet, and it must consist of \n' +
+          'only underscores,lowercase alphabets, numbers,\n' +
+          'and no spaces in between characters.');
+          done();
+        });
+    });
+
+    it('it should not create a user if the username contains an uppercase \n' +
+    'character', (done) => {
+      const user = {
+        username: 'Test',
```

Comment:
I don't think this is necessary you can converted input to lowercase at the backend before saving